src/bundle_builder.rs

```diff
@@ -0,0 +1,167 @@
+use rand::{distributions::Standard, Rng};
+
+use ethers::prelude::*;
+use ethers_flashbots::BundleRequest;
+
+use eyre::Result;
+
+/// 1 kilobyte = 1024 bytes
+const KB: usize = 1024;
+
+/// Arbitrarily chosen number to cover for nonce+from+to+gas price size in a serialized
+/// transaction.  TODO: get the actual overhead from the signing, etc. to pack more fully
+const TRIM_BYTES: usize = 300;
+
+#[tracing::instrument(skip_all, name = "construct_bundle")]
+fn construct_tx(
+    chain_id: u64,
+    address: Address,
+    receiver: Address,
+    data_size: usize,
+    gas_price: U256,
+) -> ethers::prelude::TransactionRequest {
+    // Craft the transaction.  data_size is in bytes
+    let blob = generate_random_data(data_size);
+
+    let tx = TransactionRequest::new()
+        .chain_id(chain_id)
+        .value(0)
+        .from(address)
+        .to(receiver)
+        .data(blob)
+        .gas_price(gas_price);
+
+    return tx;
+}
+
+async fn get_signed_tx<M: Middleware>(
+    chain_id: u64,
+    address: H160,
+    receiver: H160,
+    chunk: usize,
+    gas_price: U256,
+    provider: M,
+    nonce: U256,
+) -> Result<Bytes>
+where
+    M::Error: 'static,
+{
+    let mut tx = construct_tx(chain_id, address, receiver, chunk, gas_price);
+    let gas_per_tx = provider.estimate_gas(&tx.clone().into(), None).await?;
+    tracing::debug!("tx cost {} gas", gas_per_tx);
+    let blob_len = tx.data.as_ref().map(|x| x.len()).unwrap_or_default();
+    tracing::debug!("created a {blob_len} byte tx from {chunk} size data");
+
+    // apply nonce and tx gas limit
+    tx.nonce = Some(nonce);
+    tx.gas = Some(gas_per_tx);
+
+    // make into typed tx for the signer
+    let tx = tx.into();
+    let sender = provider.default_sender().unwrap_or_default();
+    let signature = provider.sign_transaction(&tx, sender).await?;
+    tracing::debug!("sender is {sender} , signature = {signature}");
+    let rlp = tx.rlp_signed(&signature);
+
+    println!("{}", serde_json::to_string(&tx)?);
+
+    // ad hoc test: submit directly
+    let tx = provider.send_transaction(tx, None).await?.await?;
```

Comment:
added this line because bundle submission was giving 

`Error: Client error: {"error":"unable to decode txs"}`

Reply:
need to figure out whats wrong with the generated transaction, will compare against master